<?xml version="1.0"?>

<launch>
    <arg name="joy_dev" default="/dev/input/js1"/>
    <arg name="joy_config" default="xbox360"/>
    <arg name="teleop_config" default="$(find ndt_gslam)/config/$(arg joy_config).yaml"/>

    <node pkg="joy" type="joy_node" name="joy_node">
        <param name="dev" value="$(arg joy_dev)"/>
        <param name="deadzone" value="0.2"/>
        <param name="autorepeat_rate" value="20"/>
    </node>

    <rosparam file="$(arg teleop_config)" command="load"/>
    <node pkg="joy_teleop" type="joy_teleop.py" name="joy_teleop" respawn="true" output="screen"></node>

    <!-- Start Gazebo with wg world running in (max) realtime -->
    <include file="$(find hector_gazebo_worlds)/launch/start.launch">
        <arg name="world" value="$(find ndt_gslam)/worlds/mff_2f.world"/>
        <arg name="gui" value="true"/>
    </include>

    <!-- Spawn simulated quadrotor uav -->
    <include file="$(find hector_quadrotor_gazebo)/launch/spawn_quadrotor_with_laser.launch">
        <!-- <arg name="z" value="0.75"/> -->
        <!-- <arg name="model" value="$(find hector_quadrotor_description)/urdf/quadrotor_hokuyo_utm30lx.gazebo.xacro"/>
    <arg name="controllers" value="
        controller/attitude
        controller/velocity
        controller/position
        "/> -->
    </include>

    <!-- Start SLAM system -->

    <node pkg="ndt_gslam" type="graph_slam2d" name="graph_slam2d" respawn="true" output="screen">
        <!-- <param name="tf_prefix" value="p3dx_0"/> -->
        <param name="robot_base_frame_id" value="base_footprint"/>
        <!-- incomming odometry  tf frame_id without tf_prefix -->
        <param name="odom_farme_id" value="world"/>
        <!-- calculated map frame_id name -->
        <param name="map_farme_id" value="map"/>

        <!-- <remap from="/scan" to="/base_scan"/> -->
        <!-- ODOM, POSE, default:NON -->
        <param name="subscribe_mode" value="NON"/>

        <param name="scanmatch_window_radius" type="double" value="10.0"/>
        <param name="node_gen_distance" type="double" value="2.0"/>
        <param name="loop_max_distance" type="double" value="20.0"/>
        <param name="loop_min_distance" type="double" value="20.0"/>
        <param name="loop_score_threshold" type="double" value="0.6"/>

        <param name="serialize_graph" type="bool" value="true"/>
        <param name="z_min" type="double" value="0.7"/>
        <param name="z_max" type="double" value="0.8"/>
    </node>

    <!-- <include file="$(find hector_mapping)/launch/mapping_default.launch">
   <arg name="odom_frame" value="world"/>
 </include> -->

    <!-- Start rviz visualization with preset config -->
    <!-- <node pkg="rviz" type="rviz" name="rviz" args="-d $(find ndt_gslam)/config/quadrocopter.rviz"/> -->

</launch>
